test/scale.test.ts

```diff
@@ -75,4 +83,70 @@ describe('scale', () => {
       }
     });
   });
+
+  describe('generateScaleTypeIndexKey', () => {
```

Comment:
These two methods are internal implementation of `getSupportedScaleType`, but not what users will use directly. 

Better test `getSupportedScaleType`  and make these two methods private (Don't export them.) 

In a way, even if we change the implementation, the test should preserve the result of `getSupportedScaleType`.